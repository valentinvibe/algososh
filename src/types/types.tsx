import {ElementStates} from './element-states';

export interface ICircleElement {
  adding?: boolean;
  deleting?: boolean;
  withoutArrow?: boolean;
  tail?: string;
  head?: string;
  char?: string | null;
  extraCircle?: {
    char: string;
  };
  state: ElementStates;
}

export type TBar = {
  num: number;
  state: ElementStates;
};

export type TElStateList = {
  isAddingToHead: boolean;
  isAddingToTail: boolean;
  isDeletingFromHead: boolean;
  isDeletingFromTail: boolean;
  isAddingByIdx: boolean;
  isDeletingByIdx: boolean;
};