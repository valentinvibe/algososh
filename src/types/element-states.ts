export enum ElementStates {
  Default = 'default',
  Changing = 'changing',
  Modified = 'modified',
  Selected = 'selected'
}