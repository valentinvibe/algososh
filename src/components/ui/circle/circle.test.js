import renderer from 'react-test-renderer';

import { Circle } from "./circle";
import { ElementStates } from "../../../types/element-states";

describe(Circle, () => {
    it('без буквы', () => {
        const tree = renderer.create(<Circle />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    
    it('с буквами', () => {
        const letters = 'abcd';
        const tree = renderer.create(<Circle letter={letters} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    
    it('с head', () => {
        const head = 'head';
        const tree = renderer.create(<Circle head={head} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    
    it('с react-элементом в head', () => {
        const head = <Circle />;
        const tree = renderer.create(<Circle head={head} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    
    it('с tail', () => {
        const tail = 'tail';
        const tree = renderer.create(<Circle tail={tail} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('с react-елементом в tail', () => {
        const tail = <Circle />;
        const tree = renderer.create(<Circle tail={tail} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('с index', () => {
        const index = 0;
        const tree = renderer.create(<Circle index={index} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    
    it('с пропсом isSmall', () => {
        const tree = renderer.create(<Circle isSmall={true} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    
    it('в состоянии default', () => {
        const state = ElementStates.Default;
        const tree = renderer.create(<Circle state={state} />).toJSON();
        expect(tree).toMatchSnapshot();
    });


    it('в состоянии changing', () => {
        const state = ElementStates.Changing;
        const tree = renderer.create(<Circle state={state} />).toJSON();
        expect(tree).toMatchSnapshot();        
    });

    it('в состоянии modified', () => {
        const state = ElementStates.Modified;
        const tree = renderer.create(<Circle state={state} />).toJSON();
        expect(tree).toMatchSnapshot();        
    });
})