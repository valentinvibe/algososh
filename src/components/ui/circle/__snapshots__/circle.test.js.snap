// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle без буквы 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle в состоянии changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle в состоянии default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle в состоянии modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle с head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    head
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle с index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    0
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Circle с react-елементом в tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Circle с react-элементом в head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle с tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    tail
  </div>
</div>
`;

exports[`Circle с буквами 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      abcd
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle с пропсом isSmall 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
